### PR TITLE
Added V.0.0.1 schema extractor

### PR DESCRIPTION
**Pull request by nokuyo**
V 0.0.1 schema extractor. Note that validation will happen within the Rust compiler so will not be in PR. Also do not have any checks for edge cases, currently just extracts correct config ts files.

### CODE REVIEW

**Review by bens-schreiber — CHANGES_REQUESTED** [review]
Needs some changes, make sure it works with the Rust portion. There aren't any tests yet so write some with Jest or whatever library you want to use.

Also add to the GitHub ci typescript specific tests, probably being:
* formatter
* type checker `tsc`
* build
* tests

File: src/schema-extractor/cloesce-ts/src/extract.ts [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+  if (!modelsDir) {
+    throw new Error(`Could not find models directory "models-cloesce" in ${cwd}`);
+  }
+
+  // ONLY this file suffix, same reason as above
+  const extPattern = opts.extPattern ?? /\.cloesce\.ts$/;
+  const files = walkFiles(modelsDir, extPattern);
+  if (files.length === 0) {
+    throw new Error(`No ".cloesce.ts" files found in "${modelsDir}"`);
+  }
```
[the remark this answers is not in the document]
Comment (nokuyo) [reply src/schema-extractor/cloesce-ts/src/extract.ts:95]: Fair, we can scrap the file lookup feature going forward and just assume any file within the models-cloesce folder is to be turned into a schema.

[the remark this answers is not in the document]
Comment (bens-schreiber) [reply src/schema-extractor/cloesce-ts/src/extract.ts:95]: either that or we just search for any `*.cloesce.ts` regardless of directory, either seems fine

[the remark this answers is not in the document]
Comment (bens-schreiber) [reply src/schema-extractor/cloesce-ts/src/extract.ts:95]: the latter would allow people to do any folder structure they want which might be better?

File: src/schema-extractor/cloesce-ts/tests/snap/dog.cloesce.ts
[the remark this answers is not in the document]
Comment (nokuyo) [reply src/schema-extractor/cloesce-ts/tests/snap/dog.cloesce.ts:1]: The logic is that both classes cover the entirety of the milestone v.0.0.1, the static methods put patch and del arent apart of the current milestone that I defined. I dont see what other edge cases qe could test, maybe just the user defining a single attribute or static method?

File: src/schema-extractor/cloesce-ts/src/extract.ts [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+  const out: string[] = [];
+  for (const entry of fs.readdirSync(root, { withFileTypes: true })) {
+    const full = path.join(root, entry.name);
+    if (entry.isDirectory()) {
+      if (!IGNORE_DIRS.has(entry.name)) out.push(...walkCloesceFiles(full));
+    } else if (entry.isFile() && /\.cloesce\.ts$/i.test(entry.name)) {
+      out.push(full);
+    }
+  }
+  return out;
```
[the remark this answers is not in the document]
Comment (bens-schreiber) [reply src/schema-extractor/cloesce-ts/src/extract.ts:43]: if you want to have the`npm cloesce` command take in a src dir thats not a bad idea for the time being either

File: src/schema-extractor/cloesce-ts/tests/snap/schema.json [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+                    {
+                        "name": "middle_name",
+                        "type": "string",
+                        "nullable": true
+                    }
+                ],
+                "methods": [
+                    {
+                        "name": "foo",
+                        "static": false,
```
Comment (bens-schreiber) [inline src/schema-extractor/cloesce-ts/tests/snap/schema.json:29]: btw we have to use `is_static` because `static` is a reserved keyword in rust

File: src/schema-extractor/cloesce-ts/src/types.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
+export interface D1Db {
+}
+
+export type Handler = (db: D1Db, req: Request, ...args: any[]) => Promise<Response>;
```
[the remark this answers is not in the document]
Comment (nokuyo) [reply src/schema-extractor/cloesce-ts/src/types.ts:4]: Nah we just passing in that it uses a d1 database. I think we should pass the db type into the static type functions alongside R2 bucket types (when we get to it) for the sake of clarity. It's a little redundant due to the D1 decorator but still, think it opens the door for more types down the road.

**Review by bens-schreiber — APPROVED** [review]
